x-pack/plugins/fleet/server/services/actions/utils.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { KueryNode } from '@kbn/es-query';
+import { get } from 'lodash';
+
+const getFieldType = (
+  key: string | undefined | null,
+  indexMappings: FleetActionsIndexMapping
+): string => {
+  const mappingKey = `properties.${key}.type`;
+  return key != null ? get(indexMappings, mappingKey) : '';
+};
+
+export const isFieldDefined = (
+  indexMappings: FleetActionsIndexMapping | FleetActionsResultsIndexMapping,
+  key: string
+): boolean => {
+  const mappingKey = 'properties.' + key;
+  return typeof get(indexMappings, mappingKey) !== 'undefined';
+};
+
+export const hasFieldKeyError = (
+  key: string | null | undefined,
+  fieldTypes: string[],
+  indexMapping: FleetActionsIndexMapping | FleetActionsResultsIndexMapping,
+  indexType: IndexType
+): string | null => {
+  const allowedKeys = Object.keys(indexMapping.properties).join();
+  if (key == null) {
+    return `The key is empty and should be one of [${allowedKeys}]`;
+  }
+  if (key) {
+    const allowedFieldTypes = fieldTypes.every((type) =>
+      indexType === 'actions'
+        ? ALLOWED_FLEET_ACTIONS_FIELD_TYPES.includes(type)
+        : ALLOWED_FLEET_ACTIONS_RESULTS_FIELD_TYPES.includes(type)
+    );
+    const indexName = indexType === 'actions' ? '.fleet-actions' : '.fleet-actions-results';
```

Comment:
Are there `const` for these index names? if so, I suggest using those instead of hardcoding them here.

Reply:
oh yeah there are `const`s for those. Will use those.

Reply:
done 7b2bb27ae9f94cc74442498a55b28a8f407c692c